src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseDetailsPageE2ETest.java

```diff
@@ -0,0 +1,126 @@
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.InstructorCourseDetailsPage;
+import teammates.e2e.util.TestProperties;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_COURSE_DETAILS_PAGE}.
+ */
+public class InstructorCourseDetailsPageE2ETest extends BaseE2ETestCase {
+    private StudentAttributes student;
+    private CourseAttributes course;
+
+    private String fileName;
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCourseDetailsPageE2ETest.json");
+        student = testData.students.get("charlie.tmms@ICDetailsE2eT.CS2104");
+        if (!TestProperties.isDevServer()) {
+            student.email = TestProperties.TEST_STUDENT1_ACCOUNT;
+        }
+
+        removeAndRestoreDataBundle(testData);
+        course = testData.courses.get("ICDetailsE2eT.CS2104");
+        fileName = "/" + course.getId() + "_studentList.csv";
+    }
+
+    @BeforeClass
+    public void classSetup() {
+        deleteDownloadsFile(fileName);
+    }
+
+    @Test
+    public void allTests() {
+        AppUrl detailsPageUrl = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_DETAILS_PAGE)
+                .withUserId(testData.instructors.get("ICDetailsE2eT.instr").googleId)
+                .withCourseId(course.getId());
+        InstructorCourseDetailsPage detailsPage = loginAdminToPage(detailsPageUrl, InstructorCourseDetailsPage.class);
+
+        ______TS("verify loaded details");
+        InstructorAttributes[] instructors = {
+                testData.instructors.get("ICDetailsE2eT.instr"),
+                testData.instructors.get("ICDetailsE2eT.instr2"),
+        };
+        StudentAttributes[] students = {
+                testData.students.get("alice.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("charlie.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("benny.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("danny.tmms@ICDetailsE2eT.CS2104"),
+        };
+
+        verifyCourseDetails(detailsPage, course, instructors, students);
+        detailsPage.verifyNumStudents(4);
+        detailsPage.verifyStudentDetails(students);
+
+        ______TS("send invite");
+        detailsPage.sendInvite(student);
+
+        detailsPage.verifyStatusMessage("An email has been sent to " + student.getEmail());
+        String expectedEmailSubject = "TEAMMATES: Invitation to join course ["
+                + course.getName() + "][" + course.getId() + "]";
+        verifyEmailSent(student.getEmail(), expectedEmailSubject);
+
+        ______TS("remind all students to join");
+        detailsPage.remindAllToJoin();
+
+        detailsPage.verifyStatusMessage("Emails have been sent to unregistered students.");
+        verifyEmailSent(student.getEmail(), expectedEmailSubject);
```

Comment:
Here, we also check email sent for just one student. Shouldn't we check all unregistered students get an email (which I think is all for this case)?

Reply:
On top of the same reasoning as before (just one representative is enough for an E2E test), there is another more fundamental reason here. Checking email sent in production server requires actual account and even within the actual account some setups are required. It's not feasible to set up more accounts than required just for the sake of this.

Reply:
Yes that is true! I agree